frontend/packages/dev-console/src/components/topology2/Topology.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import {
+  TopologyView,
+  TopologyControlBar,
+  createTopologyControlButtons,
+  defaultControlButtonsOptions,
+} from '@patternfly/react-topology';
+import Visualization from '@console/topology/src/Visualization';
+import VisualizationWidget from '@console/topology/src/VisualizationWidget';
+import { ElementEntity, isNodeEntity, Model } from '@console/topology/src/types';
+import {
+  SELECTION_EVENT,
+  SelectionEventListener,
+} from '@console/topology/src/behavior/useSelection';
+import BaseNodeEntity from '@console/topology/src/entities/BaseNodeEntity';
+import * as _ from 'lodash';
+import TopologySideBar from '../topology/TopologySideBar';
+import { TopologyDataModel, TopologyDataObject } from '../topology/topology-types';
+import TopologyResourcePanel from '../topology/TopologyResourcePanel';
+import TopologyApplicationPanel from '../topology/TopologyApplicationPanel';
+import { topologyModelFromDataModel } from './topology-utils';
+import layoutFactory from './layoutFactory';
+import widgetFactory from './widgetFactory';
+
+export interface TopologyProps {
+  data: TopologyDataModel;
+}
+
+const graphModel: Model = {
+  graph: {
+    id: 'g1',
+    type: 'graph',
+    layout: 'Force',
+  },
+};
+
+const Topology: React.FC<TopologyProps> = ({ data }) => {
+  const visRef = React.useRef<Visualization | null>(null);
+  const [model, setModel] = React.useState<Model>();
+  const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
+  const [selectedEntity, setSelectedEntity] = React.useState<ElementEntity | undefined>();
+  const { topology } = data;
+
+  const onSelection = (ids: string[]) => {
+    setSelectedIds(ids);
+  };
+
+  if (!visRef.current) {
+    visRef.current = new Visualization();
+
+    visRef.current.registerLayoutFactory(layoutFactory);
+    visRef.current.registerWidgetFactory(widgetFactory);
+    visRef.current.addEventListener<SelectionEventListener>(SELECTION_EVENT, onSelection);
```

Comment:
Can remove `onSelection` and do the following:
```suggestion
    visRef.current.addEventListener<SelectionEventListener>(SELECTION_EVENT, setSelectedIds);
```